wgpu-hal/src/gles/egl.rs

```diff
@@ -149,8 +154,10 @@ fn open_x_display() -> Option<DisplayOwner> {
     log::debug!("Loading X11 library to get the current display");
     unsafe {
         let library = find_library(&["libX11.so.6", "libX11.so"])?;
-        let func: libloading::Symbol<XOpenDisplayFun> =
-            library.get(c"XOpenDisplay".to_bytes()).unwrap();
+        let func: libloading::Symbol<XOpenDisplayFun> = library
+            .get(c"XOpenDisplay".to_bytes())
+            .inspect_err(|err| log::error!("Failed to load XOpenDisplay: {err:?}"))
```

Comment:
feels like we should have an error category for failing to load methods which we can propagate. But this is already strictly better 🤷

Reply:
libloading can absolutely fail querying a function for all sort of reasons. It's highly unlikely but I wouldn't dismiss it as can never happen.